src/Searchbar.vue

```diff
@@ -21,29 +21,42 @@ export default {
       if (!this.data) {
         return undefined;
       }
+
+      function calculateMatchNum(searchTarget, regexes) {
+        return regexes.reduce((matchNum, regex) => (regex.test(searchTarget) ? matchNum + 1 : matchNum), 0);
+      }
+
       const matches = [];
-      const regexes = this.value.split(' ').map(searchKeyword => new RegExp(searchKeyword, 'i'));
+      const regexes = this.value.split(' ').filter(searchKeyword => searchKeyword !== '')
+        .map(searchKeyword => new RegExp(searchKeyword, 'i'));
       this.data.forEach((entry) => {
         const { headings, src, title } = entry;
         const keywords = entry.keywords || '';
-        let searchTarget = [title].concat(keywords).join(' ');
-        const isMatchingPage = regexes.every(regex => regex.test(searchTarget));
-        if (isMatchingPage) {
-          matches.push(entry);
-        }
-        Object.entries(headings).forEach(([id, text]) => {
-          searchTarget = [title].concat(keywords).concat(text).join(' ');
-          if (isMatchingPage || regexes.every(regex => regex.test(searchTarget))) {
-            matches.push({
-              heading: { id, text },
-              keywords,
-              src,
-              title,
-            });
+        let searchTarget = [title].concat(keywords).concat(Object.values(headings)).join(' ');
+        let matchNum = calculateMatchNum(searchTarget, regexes);
+
+        if (matchNum) {
```

Comment:
Might be good to be a bit more explicit in the condition (i.e. `if (matchNum > 0)`).